.travis.yml

```diff
@@ -1,6 +1,10 @@
 language: cpp
 sudo: false
 
+git:
+  depth: 10
+  quiet: true
```

Comment:
I'll have to re-check how depth works, if it is passed to submodules this isn't going to work well with branches (submodule commits won't be reached). I think I've tried it in the past, but I don't really remember it.

As for quiet, I'd also say no, that hides important information for when problems occur.